core/src/test/java/hudson/util/AtomicFileWriterTest.java

```diff
@@ -124,4 +142,72 @@ public void badPath() throws Exception {
         }
     }
+
+    @Issue("JENKINS-48407")
+    @Test
+    public void preserveSpecificPermissions() throws IOException, InterruptedException {
+
+        // given
+        final Set<PosixFilePermission> givenPermissions = EnumSet.of(OWNER_READ,
+                                                                     OWNER_WRITE,
+                                                                     OWNER_EXECUTE,
+                                                                     GROUP_READ);
+
+        final Set<PosixFilePermission> notGivenPermissions = EnumSet.of(GROUP_WRITE,
+                                                                        GROUP_EXECUTE,
+                                                                        OTHERS_READ,
+                                                                        OTHERS_WRITE,
+                                                                        OTHERS_EXECUTE);
+
+        final File newFilePath = tmp.newFile();
+        Files.setPosixFilePermissions(newFilePath.toPath(), givenPermissions);
+        runAndCheckPermissions(newFilePath.toPath(), givenPermissions, notGivenPermissions);
+    }
+
+    /**
+     * We create a temporary dir, so that then we can use Files.createFile() instead of Files.createTempDir().
+     * Because createFile will create with default permissions, when NIO's createTempDir() will with 0600 on L
+     *
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    @Issue("JENKINS-48407")
+    @Test
+    public void defaultPermissionsOnNonExistingFile() throws IOException, InterruptedException {
+
+        assertNotNull(Jenkins.DEFAULT_FILE_PERMISSIONS);
+        // given
+        final File newFilePath = tmp.newFolder("temporary");
+        final Path newFile = Files.createFile(Paths.get(newFilePath.getAbsolutePath(), "newFile"));
+
+        final Set<PosixFilePermission> givenPermissions = Files.getPosixFilePermissions(newFile);
+        assertEquals(Jenkins.DEFAULT_FILE_PERMISSIONS, givenPermissions);
+
+        final Set<PosixFilePermission> notGivenPermissions =
+                Arrays.stream(PosixFilePermission.values())
+                        .filter(permission -> !givenPermissions.contains(permission))
+                        .collect(Collectors.toSet());
+        runAndCheckPermissions(newFile, givenPermissions, notGivenPermissions);
+    }
+
+    private void runAndCheckPermissions(Path newFilePath, Set<PosixFilePermission> givenPermissions, Set<PosixFilePermission> notPermissions) throws IOException {
+        // when
+        AtomicFileWriter w = new AtomicFileWriter(newFilePath, StandardCharsets.UTF_8);
+        w.write("whatever");
+        w.commit();
+
+        // then
+        assertFalse(w.getTemporaryPath().toFile().exists());
+        assertTrue(newFilePath.toFile().exists());
+
+        final Set<PosixFilePermission> posixFilePermissions = Files.getPosixFilePermissions(newFilePath);
```

Comment:
posix on a non posix system will blow up with an exception that is unhandled.

https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#getPosixFilePermissions-java.nio.file.Path-java.nio.file.LinkOption...-

> Throws:
UnsupportedOperationException - if the associated file system does not support the PosixFileAttributeView